object-api-multipart.go

```diff
@@ -347,11 +380,22 @@ func (o objectAPI) PutObjectPart(bucket, object, uploadID string, partID int, si
 	if size > 0 {
 		if _, e = io.CopyN(multiWriter, data, size); e != nil {
 			safeCloseAndRemove(fileWriter)
-			if e == io.ErrUnexpectedEOF {
+			if e == io.ErrUnexpectedEOF || e == io.ErrShortWrite {
 				return "", probe.NewError(IncompleteBody{})
 			}
 			return "", probe.NewError(e)
 		}
+		// Allocating a byte to read from the reader.
+		// This validate whether more data exists even after 'size' number of bytes.
+		singleByte := make([]byte, 1)
+		singleByteBuffer := bytes.NewBuffer(singleByte)
+		// Reader shouldn't have more data what mentioned in size arguement.
+		// reading one more byte from the reader to validate it.
+		// expected to fail, success validates existance of more data in the reader.
+		if _, e = io.CopyN(singleByteBuffer, data, 1); e == nil {
+			safeCloseAndRemove(fileWriter)
+			return "", probe.NewError(UnExpectedDataSize{Message: fmt.Sprintf("Contains more data then specified size of %d bytes.", size)})
```

Comment:
The message should go into the Error() string in the definition itself. Not here.